frappe/core/doctype/user_type/user_type.py

```diff
@@ -114,7 +114,7 @@ def add_select_perm_doctypes(self):
 		self.select_doctypes = []
 
 		select_doctypes = []
-		user_doctypes = tuple([row.document_type for row in self.user_doctypes])
+		user_doctypes = tuple(row.document_type for row in self.user_doctypes)
```

Comment:
We can leave this as a generator too, right? Let Python do its thing while iterating over it on L119

Reply:
yeah, I kept these for readability sake 🤔  generators aren't that intuitive while reading at least.

If someone changes / decides to use a generator by direct indexing they'll face an error. 🤔

Reply:
> yeah, I kept these for readability sake 🤔 generators aren't that intuitive while reading at least.

Maybe the case, or just using them wherever apt will increase adaptability and reduce this issue over time 🤔. However, in this case, the list comprehension might've been better.

```python
user_doctypes = [row.document_type for row in self.user_doctypes]
```

over

```python
user_doctypes = tuple(row.document_type for row in self.user_doctypes)
```
---

> If someone changes / decides to use a generator by direct indexing they'll face an error. 🤔

If this was attached to an instance, sure but I don't think you need to worry about that since `user_doctypes` isn't supposed to be used in any other scope.

Reply:
Yeah, I don't know why it was made a tuple 🙃